infra/scripts/test-docker-compose.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+set -e
+
+echo "
+============================================================
+Running Docker Compose tests with pytest at 'tests/e2e'
+============================================================
+"
+
+export PROJECT_ROOT_DIR=$(git rev-parse --show-toplevel)
+export COMPOSE_INTERACTIVE_NO_CLI=1
+
+# Create Docker Compose configuration file
+cd ${PROJECT_ROOT_DIR}/infra/docker-compose/
+cp .env.sample .env
+
+# Start all Docker Compose containers
+docker-compose -f docker-compose.yml -f docker-compose.online.yml up -d
+
+# Get Docker Container IP address
+export JUPYTER_DOCKER_CONTAINER_IP_ADDRESS=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' feast_jupyter_1)
+
+
+# Print Jupyter container information
+docker inspect feast_jupyter_1
+docker logs feast_jupyter_1
+
+# Wait for Jupyter Notebook Container to come online
+timeout 5m bash -c 'until [ $(curl -s ${JUPYTER_DOCKER_CONTAINER_IP_ADDRESS}:8888/tree | grep notebook | wc -l) -gt 0 ]; do echo waiting for Jupyter to start...; echo; sleep 5; done'
```

Comment:
@khorshuheng @davidheryanto must say that I quite like `timeout` as opposed to sleep. As long as you can create a nice little logical test. We can replace sleeps with this maybe.

Reply:
Maybe we can replace this with [`wait-for-it.sh`](https://github.com/vishnubob/wait-for-it)? So that we don't have to nest a while loop here. 
Its a pure bash script that blocks until a specific port comes alive, so in this case we can wait for `8888` to come up:
```
./wait-for-it.sh ${JUPYTER_DOCKER_CONTAINER_IP_ADDRESS}:8888 --timeout=300 
```
Although I'm not sure if the explicit check for the notebook  
(`curl -s ${JUPYTER_DOCKER_CONTAINER_IP_ADDRESS}:8888/tree | grep notebook | wc -l) -gt 0 `)  
is necessary to after `8888` comes up.

Reply:
Oooh, this is nice!